### PR TITLE
Add double-buffering & atari support

### PR DESCRIPTION
**Pull request by mikrosk**
Hi,

a looong time ago (2010) I ported your HHeretic & HHexen to Atari (m68k, a strong CPU required of course). 14 years later, while testing my Atari SDL 1.2 port I have noticed a few areas for improvement also in these two projects (what a pleasant surprise to find them still maintained!)

Most notably I have added support for double-buffering so even slow platforms like Atari can have decent framerate and tearing-less rendering.

I can offer a similar PR for HHexen but I wanted to hear from you first.

Cheers,
Miro

### CODE REVIEW

File: base/r_main.c [hunk 1]
```diff
@@ -790,9 +790,6 @@ void R_SetupFrame(player_t *player)
 		UpdateState |= I_MESSAGES;
 	}
 
-#if defined(__WATCOMC__) && defined(_DOS)
-	destview = destscreen + (viewwindowx>>2) + viewwindowy*80;
```
Comment (sezero) [inline base/r_main.c:794]: They are 'used' in I_InitDiskFlash() of i_ibm.c in the original dos version, but it is commented out.

**Comment by sezero** [comment]
I very briefly tested, seems to run OK. I'd welcome an equivalent patch for hhexen.

File: base/r_main.c (hunk 1 above)
Comment (mikrosk) [reply base/r_main.c:794]: Indeed, I saw it too but I don't think anyone is ever going to use it. :-P

Comment (sezero) [reply base/r_main.c:794]: OK, let's keep it. It doesn't have to exist in sdl sources but keep it in other places

**Comment by sezero** [comment]
OK, I hand-picked patches and `git am`'ed them. Thanks!

**Pull request closed without merging**